src/components/Navbar/NavbarLink.tsx

```diff
@@ -28,12 +28,18 @@ export const NavbarLink: FC<NavbarLinkProps> = ({
   children,
   className,
   theme: customTheme = {},
+  onClick,
   ...props
 }) => {
-  const { theme: rootTheme } = useNavbarContext();
+  const { theme: rootTheme, isOpen, setIsOpen } = useNavbarContext();
 
   const theme = mergeDeep(rootTheme.link, customTheme);
 
+  const handleClick = (event: MouseEvent<HTMLAnchorElement>) => {
+    setIsOpen(!isOpen);
```

Comment:
I'd rather be explicit to closing the navbar instead of toggle-ing it (true/false).

Suggestion:
```tsx
const handleClick = (event: MouseEvent<HTMLAnchorElement>) => {
  setIsOpen(false);
  onClick?.(event);
};
```

This also makes the `isOpen` value redundant and can be removed.

Reply:
made the changes you mentioned!..